src/helper/helper.js

```diff
@@ -429,3 +429,12 @@ exports = {
   isString_,
   merge_,
 };
+
+// Closure compiler advanced optimizations will compile get and flatten
+// normally, but this is only because they are library functions somewhere and
+// it may not be intended behavior. Just in case, I add them here
+// see https://groups.google.com/g/closure-compiler-discuss/c/Z5rorPYR0m4/m/7vijpvpqCgAJ
```

Comment:
Honestly, I'm not convinced that this behavior is a bug as closure does the best it can while guessing whether some code is used do dead code could be removed.. unfortunately in the javascript world sometimes it could be a mystery so more conservative approach must be taken...

I wouldn't mention in the comment that closure will compile get and flatten normally as it could easily change when closure becomes "smarter" about what method is used (if it's even possible).

That said, I'd just mention that this code isn't used directly by DataLayerHelper so in order to preserve it in the compiled binary it needs to be explicitly added to the prototype (discuss link is super useful as well)


Overall, great job figuring this out!

Reply:
I don't think this closure behavior has to do with the compiler noticing if the method is used, it seems to be based only on the method name. I suspect you are right about behavior changing in the future though, (especially since it already produces different results at https://closure-compiler.appspot.com/home ). So I updated the comment